### ISSUE TITLE
Jetpack activation causes error.log clutter: PHP Warning:  call_user_func_array() expects parameter 1 to be a valid callback

### ISSUE THREAD

**Issue by s-a-s-k-i-a** [Automattic/jetpack#9749]
Hello there!
As soon as i activate Jetpack plugin I am seeing this php warning cluttering my error.log (with wp debug mode enabled):

`PHP Warning:  call_user_func_array() expects parameter 1 to be a valid callback, no array or string given in /var/www/domain.com/htdocs/wp-includes/class-wp-hook.php on line 286`

Call Stack
--

```
1 | 0.0003 | 400296 | {main}( ) | .../index.php:0
2 | 0.0006 | 400664 | require( '/var/www/domain.com/htdocs/wp-blog-header.php' ) | .../index.php:17
3 | 0.0010 | 401608 | require_once( '/var/www/domain.com/htdocs/wp-load.php' ) | .../wp-blog-header.php:13
4 | 0.0012 | 402424 | require_once( '/var/www/domain.com/wp-config.php' ) | .../wp-load.php:42
5 | 0.0023 | 404288 | require_once( '/var/www/domain.com/htdocs/wp-settings.php' ) | .../wp-config.php:54
6 | 1.1747 | 16926296 | do_action( ) | .../wp-settings.php:450
7 | 1.1747 | 16926672 | WP_Hook->do_action( ) | .../plugin.php:453
8 | 1.1747 | 16926672 | WP_Hook->apply_filters( ) | .../class-wp-hook.php:310
9 | 1.3416 | 19881264 | Jetpack_Carousel->init( ) | .../class-wp-hook.php:286
10 | 1.3417 | 19881264 | Jetpack_Carousel->maybe_enable_jp_carousel_single_images_media_file( ) | .../jetpack-carousel.php:42
11 | 1.3417 | 19881264 | apply_filters( ) | .../jetpack-carousel.php:125
12 | 1.3417 | 19881664 | WP_Hook->apply_filters( ) | .../plugin.php:203

```

I narrowed it down to jetpack being the culprit by de- and reactivating all plugins. This php warning only shows when Jetpack is activated.

These are my specs:
WordPress 4.9.6
Jetpack 6.2.1

server:
nginx
php 7.2.x
maria db

Any ideas?

**Comment by designsimply** [comment]
I noticed the stack trace you provided references the carousel, so I enabled that feature in Jetpack and published a post with a gallery and viewed the carousel and then tested with Jetpack only installed (WP 4.9.6, Jetpack 6.2.1, PHP 7.2) and was unable to reproduce the PHP warning you referenced.

> As soon as i activate Jetpack plugin I am seeing this php warning cluttering my error.log

I saw that the last thing in the stack trace you provided points to jetpack-carousel.php:125 which applies the `jp_carousel_load_for_images_linked_to_file` filter and I don't see Jetpack actively using that so it's possible another plugin has made a mistake or conflicts with that hook.

If you activate Jetpack *only* does the error go away? 

What other plugins are activated in addition to Jetpack when you see the PHP warning?
